src/main/java/com/scalyr/integrations/kafka/Event.java

```diff
@@ -169,18 +187,48 @@ public int estimatedSerializedBytes() {
       return estimatedSizeBytes;
     }
 
-    int size = getMessage() == null ? 0 : getMessage().length();
+    int size = Strings.isNullOrEmpty(getMessage()) ? 0 : estimateEscapedStringSize(getMessage());
     size += getTopic().length();
     size += EVENT_SERIALIZATION_OVERHEAD_BYTES;
 
     if (getAdditionalAttrs() != null) {
       size += getAdditionalAttrs().entrySet().stream()
-        .mapToInt(entry -> entry.getKey().length() + (entry.getValue() == null ? 0 : entry.getValue().toString().length())).sum();
+        .mapToInt(entry -> entry.getKey().length() + (entry.getValue() == null ? 0 : estimateEscapedStringSize(entry.getValue().toString()))).sum();
     }
     estimatedSizeBytes = size;
     return estimatedSizeBytes;
   }
 
+  /**
+   * Estimate the escaped String length for JSON strings.
+   * JSON payload requires extra escaping which needs to be taken into the size estimate.
+   *
+   * If it is not JSON, then use the String length as the estimate.
+   * This trades off some inaccuracies where the String may contain quotable characters for faster performance.
+   * @return Estimated escaped String size
+   */
+  private int estimateEscapedStringSize(String s) {
+    if (s == null) {
+      return 0;
+    }
+    if (s.length() < 2) {
+      return s.length();
+    }
+
+    if (s.charAt(0) == '{' && s.charAt(s.length() - 1) == '}') {
```

Comment:
Would this still work if JSON is already escaped? Or would that cause issues on the server side so it's not something we need to handle.

IIRC, I've seen some prescaped events which appeared to be correctly handled on the server side, but not sure if it matters or it's relevant here.

Reply:
We're estimating payload size and not mutating the original payload.  Pre-escaped json throws off the count some.  e.g.
`{"req": "{\"jobId\":\"5eef5d97dc0e82000a8b6a15\"}` gets serialized in the add events payload to `{\"req\": \"{\\\"jobId\\\":\\\"5eef5d97dc0e82000a8b6a15\\\"}` resulting in a 11 character increase.  Notice it doesn't double-escape the `\"` escape.  If we just count escapable characters, we get 22 instead of 11.

It shouldn't have any effect on the server.  The server doesn't undo pre-escaped JSON.  i.e. It displays `{"req": "{\"jobId\":\"5eef5d97dc0e82000a8b6a15\"}` as `{"req": "{\"jobId\":\"5eef5d97dc0e82000a8b6a15\"}`.